mne/utils/docs.py

```diff
@@ -2494,9 +2494,9 @@ def _reflow_param_docstring(docstring, has_first_line=True, width=75):
 ] = """\
 **method_kw
     Additional keyword arguments passed to the spectral estimation
-    function (e.g., ``n_fft, n_overlap, n_per_seg, average, window``
+    function (e.g., ``n_fft, n_overlap, n_per_seg, average, window, remove_dc``
     for Welch method, or
-    ``bandwidth, adaptive, low_bias, normalization`` for multitaper
+    ``bandwidth, adaptive, low_bias, normalization, remove_dc`` for multitaper
```

Comment:
if `remove_dc` is a common kwarg for both welch and multitaper, then it should probably become part of the main signature of `compute_psd` rather than being a `**method_kwarg` (which is meant to handle *only* the kwargs that differ between methods)

Reply:
I was thinking about that too. If we put that option outside of method_kwarg, I would need to add it almost "everywhere" in the time_frequency classes/methods, all the constructors, etc... Was wondering if there would be a way to avoid that...

Reply:
```console
$ git grep method_kw_psd
mne/epochs.py:2364:        %(method_kw_psd)s
mne/epochs.py:2455:        %(method_kw_psd)s
mne/evoked.py:1059:        %(method_kw_psd)s
mne/evoked.py:1151:        %(method_kw_psd)s
mne/io/base.py:2170:        %(method_kw_psd)s
mne/time_frequency/spectrum.py:124:        %(method_kw_psd)s
mne/time_frequency/spectrum.py:179:        %(method_kw_psd)s Defaults to ``dict(n_fft=2048)``.
mne/time_frequency/spectrum.py:263:        %(method_kw_psd)s
mne/time_frequency/spectrum.py:1066:    %(method_kw_psd)s
mne/time_frequency/spectrum.py:1201:    %(method_kw_psd)s
mne/utils/docs.py:2493:    "method_kw_psd"
```
I count ten places. That seems doable?  It is only for spectrum classes and the compute_psd methods, not for the time-frequency classes/methods I think.

Reply:
I count 56 if you do git grep method_kw, but yeah double I guess!

Reply:
Before I start making all the changes, is there a preference of where it should come in the order of variables? Here is the init of BaseSpectrum for example:
```
    def __init__(
        self,
        inst,
        method,
        fmin,
        fmax,
        tmin,
        tmax,
        picks,
        proj,
        *,
        n_jobs,
        verbose=None,
        **method_kw,
    ):
```

Reply:
ah I see your point, it is relevant to the time-freq funcs/classes even if they don't have that particular docdict entry.  I'll need to think about how we should handle the API change then, as the Spectrum API and the time-frequency APIs are somewhat different (and the TF ones are in process of being made more similar to the spectrum one, but the transition is still WIP).

Reply:
ok i'll wait for your input before making any changes

Reply:
Hi @drammock! Checking in if you have any updates on this.

Reply:
OK, sorry it took so long to get you an answer. I think we should add the new kwarg to:

- {BaseRaw, Epochs, Evoked}.compute_psd() and {BaseSpectrum,Spectrum,EpochsSpectrum}.`__init__` should all get the new named param `remove_dc`
    - you'll need to pass it through to `._compute_spectra` and from there pass it into `self._psd_func`.
- {BaseRaw, Epochs, Evoked}.plot_psd() should **NOT** --- those are legacy methods and their API shouldn't change. It should still work to pass `remove_dc` --- it will get bundled into `method_kw`
- SpectrumMixin.plot_psd(), .plot_psd_topo() and .plot_psd_topomap() should **NOT** (same reason: legacy)

Hopefully that works; this is based on `git grep` and reading/reasoning about the code; I haven't tested it.  If it turns out to fail or be really messy it won't be the end of the world to incorporate `remove_dc` into the `method_kw` everywhere.

Reply:
I'll work on it now

Reply:
finished, please have a look, i tested in my notebook, all seems ok